### DISCUSSION TITLE
Mount a Local Directory to the DDEV Container

### DISCUSSION THREAD

**Discussion by bbholty1** [ddev/ddev#6802] [Q&A]
Hello, 

Iv'e seen everything from:
 ddev config --mounts="local_files:/path/to/local/files" , which is not currently not an existing config option.
to: 
creating volumes: Path/To/LocalDir:Path/To/Mount/InDdevContainer in the .ddev/config.yaml, which hasn't been successful to this point.

Here's the reason I need to be able to Mount a Local Directory to the DDEV Container...

I'm testing a drupal module that imports media files in mass quantities at a time, from any possible existing location on the server...  And currently I have no way to get to the mass quantities of media files,, which exist outside of the ddev container,  to be able to test anything within the ddev container. 

Thoughts or ideas?

Thanks a ton for your help!

-Bryan Holty

**Comment by rfay** [comment]
Hi @bbholty1 - Are you saying that you have PHP code that imports files from inside the container? Where does it put them?

If you're using mutagen, the `upload_dirs` directive tells DDEV to mount those directories from the host, so there's nothing to do. Make sure that your local files is listed there.

If you're not using `performance_mode: mutagen` then there's nothing to do, everything in your project is already bind-mounted into the container.

If you'd like to mount something else that is not already in your project, the solution in https://stackoverflow.com/a/57432155/215713 might help you.

If you do a `ddev debug test` and provide the result, I may be able to say more.

**Comment by bbholty1** [comment]
@rfay  The php code  needs to be able to import the files from outside the container to inside the container. 

That's the issue I'm struggling with and have yet to find a solution.

Thanks

**Reply by rfay** [reply]
Again, please say what directory the PHP code is going to import into. If the files are in your project, they will be available. If they're not in your project, you can use the mounts technique I linked to.

**Reply by bbholty1** [reply]
the files are not in my project. I'm attempting the mounts method now.

Thanks!

**Reply by bbholty1** [reply]
@rfay I initially had the mounts defined in the config.yaml file and was missing the services: at the top anyway. Moved it into the proper yaml file and added the services: properly and it worked.

Thanks!
